util/rom-ext-manifest-generator.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import copy
+from pathlib import Path
+
+import hjson
+from mako.template import Template
+
+from topgen.c import MemoryRegion, Name
+
+DESC = """ROM_EXT manifest generator"""
+
+USAGE = """
+  rom-ext-manifest-generator --input-dir:
+    Directory where manifest.hjson and manifest.h.tpl reside.
+
+  rom-ext-manifest-generator --output-dir:
+    Directory where manifest.hjson and manifest.h.tpl reside.
+"""
+
+
+def add_size_in_bytes(fields):
+    """Adds `size_in_bytes` entry to every parsed ROM_EXT manifest field.
+
+    fields[N]    =>    fields[N]
+      - ...              - ...
+      - size             - size
+                         - size_in_bytes
+    """
+
+    for field in fields:
+        field['size_in_bytes'] = field['size'] // 8
+
+
+def add_offsets_in_bytes(fields):
+    """Adds `offset_in_bytes` entry to every parsed ROM_EXT manifest field.
+
+    fields[N]    =>    fields[N]
+      - ...              - ...
+      - size_in_bytes    - size_in_bytes
+                         - offset_in_bytes
+    """
+
+    accumulated_offset = 0
+
+    for field in fields:
+        field['offset'] = hex(accumulated_offset)
+        accumulated_offset += field['size_in_bytes']
+
+
+def generate_cheader(fields, input_dir, output_dir):
+    """ Generates C header file from the `template_file`.
+
+    It produces a list of tuples with a field name and the `MemoryRegion`
+    object, whic is used in the `template_file`. The resulting header file is
+    placed into `output_dir`.
+    """
+
+    manifest_cheader_template_file = input_dir / 'manifest.h.tpl'
+    manifest_cheader_output_file = output_dir / 'manifest.h'
```

Comment:
It's not like this code is wrong, but these are local names for a function that generates a manifest C header. So they could probably be called just `template_file` and `output_file`, shortening lines and generally making things easier to read.

Also, I'd personally call them `template_path` and `output_path`, since they are paths, not file objects (a Python file object is a wrapper around a file descriptor, a bit like C's `FILE*`).

Reply:
Done.

Reply:
The latest push on the PR has not changed the code here. Please either leave the original reviewer to mark something done, or (if it's obviously addressed) do so yourself *after* pushing up the corresponding change. Otherwise the reviewer can't trust whether something has been changed or not.

Reply:
Fair point, will do in future.

Reply:
All the changes have been pushed now.